### DISCUSSION TITLE
Timelines for the various OS image versions?

### DISCUSSION THREAD

**Discussion by mhever** [actions/virtual-environments#9606] [General questions] [answered]
Is there a un/official timeline for the various image versions? What is the plan with ubuntu 20, do you have a date when support stops for it? Same question for Windows 19? Any plans to introduce an Ubuntu 24 one?

**Comment by shamil-mubarakshin** [comment]
Hey @mhever. There are no official timelines for image addition.
We plan to introduce Ubuntu24 and start working towards it as soon as Azure offering becomes available. It will go through Beta to GA. 
Once Ubuntu24 is in GA, Ubuntu20 deprecation process will be kicked off.
Similar is true for Windows 2019

**Reply by mhever** [reply, marked as answer]
Thanks, that gives us ample time to migrate away from Ubuntu20
